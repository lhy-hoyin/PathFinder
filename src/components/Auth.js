import { useState } from "react";
import { supabase } from "../supabaseClient";

export default function Auth() {
  const [isLoading, setLoading] = useState(false);
  const [email, setEmail] = useState("");

  const handleLogin = async e => {
    e.preventDefault();

    try {
      setLoading(true);

      const { error } = await supabase.auth.signIn({ email });
      if (error) throw error;
      alert("Check your email for the login link!");
    } catch (error) {
      alert(error.error_description || error.message);
    } finally {
      setLoading(false);
    }
  };

  return (
    <auth>
      <h2>Let's Get Started</h2>
      <form onSubmit={handleLogin}>
        <label htmlFor="email">Email</label>
        <input
          id="email"
          className="inputField"
          type="email"
          placeholder="Your email"
          value={email}
          onChange={e => setEmail(e.target.value)}
        />

        <button className="button block" aria-live="polite">
          Sign Up
        </button>

        <button className="button block" aria-live="polite">
          Login using email
        </button>
      </form>

      {isLoading ? <p>Signing up, please wait</p> : ""}
      <p>{email}</p>
    </auth>
  );
}
