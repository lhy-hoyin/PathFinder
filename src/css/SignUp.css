@import "styles.css";

button,
.button {
    color: var(--btn-color-text);
    border: var(--custom-border);
    background-color: var(--btn-color-bg-default);
    display: inline-block;
    text-align: center;
    border-radius: var(--custom-border-radius);
    padding: 0.5rem 1rem;
    cursor: pointer;
    text-align: center;
    font-size: 0.9rem;
    text-transform: uppercase;
}

    button.primary,
    .button.primary {
        background-color: var(--custom-color-brand);
        border: 1px solid var(--custom-color-brand);
    }

.frame {
    width: 100%;
    display: flex;
    position: relative;
    justify-content: center;
}

.register-new {
    width: 46%;
    display: flex;
    flex-direction: column;
    gap: 20px;
}

    .register-new > .button {
        display: flex;
        align-items: center;
        justify-content: center;
        border: none;
        background-color: #444444;
        text-transform: none !important;
        transition: all 0.2s ease;
    }

    .register-new .button:hover {
        background-color: var(--btn-color-bg-hover);
    }

    .register-new .button > .loader {
        width: 17px;
        animation: spin 1s linear infinite;
        filter: invert(1);
    }

input {
    width: 100%;
    border-radius: 5px;
    border: var(--custom-border);
    padding: 8px;
    font-size: 0.9rem;
    background-color: var(--custom-bg-color);
    color: var(--custom-color);
}

    input[disabled] {
        color: var(--custom-color-secondary);
    }



.block {
    display: block;
    width: 100%;
}

/*/
html,
body {
    --custom-font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
    --custom-bg-color: #101010;
    --custom-panel-color: #222;
    --custom-box-shadow: 0 2px 8px 0 rgba(0, 0, 0, 0.8);
    --custom-color: #fff;
    --custom-color-brand: #24b47e;
    --custom-color-secondary: #666;
    --custom-border: 1px solid #333;
    --custom-border-radius: 5px;
    --custom-spacing: 5px;
    padding: 0;
    margin: 0;
    font-family: var(--custom-font-family);
    background-color: var(--custom-bg-color);
}

* {
    color: var(--custom-color);
    font-family: var(--custom-font-family);
    box-sizing: border-box;
}
/**/