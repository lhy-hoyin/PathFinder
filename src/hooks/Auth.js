import { useState, useEffect, useContext, createContext } from "react";
import { supabase } from "../supabaseClient";

const authContext = createContext();


export function ProvideAuth({ children }) {
    const auth = useProvideAuth();
    return <authContext.Provider value={auth}>{children}</authContext.Provider>;
}

export const Auth = () => {
    return useContext(authContext);
};

function useProvideAuth() {
    const [session, setSession] = useState(null);
    const [profileInfoReady, setProfileInfoReady] = useState(false);

    // User-releated info
    const [email, setEmail] = useState(null);
    const [firstName, setFirstName] = useState(null);
    const [lastName, setLastName] = useState(null);
    const [cohort, setCohort] = useState(null);
    const [isReady, setIsReady] = useState(null);
    const [isLocked, setIsLocked] = useState(null);

    useEffect(() => {
        setSession(supabase.auth.session());
        supabase.auth.onAuthStateChange((_event, session) => {setSession();})
    }, []);

    useEffect(() => {
        getProfile();
    }, [session]);

    const getProfile = async () => {
        try {
            const user = supabase.auth.user();

            if (user == null)
                return

            setEmail(user.email);

            let { data, error, status } = await supabase
                .from('profiles')
                .select('*')
                .eq('id', user.id)
                .single()

            if (error && status !== 406)
                throw error

            if (data) {
                setFirstName(data.FirstName);
                setLastName(data.LastName);
                setCohort(data.Cohort);
                setIsReady(data.isReady);
                setIsLocked(data.isLocked);

                setProfileInfoReady(true);
            }
        } catch (error) {
            console.error(error.message);
        }
    };

    const updateProfileBasic = (fName, lName) => async e => {
        e.preventDefault();

        try {
            const user = supabase.auth.user()

            if (user == null)
                return

            // Pacakage data properly
            const updates = {
                id: user.id,
                FirstName: fName,
                LastName: lName,
                isReady: true,
                updated_at: new Date(),
            }

            let { error } = await supabase.from('profiles').upsert(updates, {
                returning: 'minimal', // Don't return the value after inserting
            })

            if (error)
                throw error
            else
                console.log("Profile Updated successfully")

        } catch (error) {
            console.error(error.message);
        }
    };

    const updateProfileAcad = (cohort) => async e => {
        e.preventDefault();

        try {
            const user = supabase.auth.user()

            if (user == null)
                return

            // Pacakage data properly
            const updates = {
                id: user.id,
                Cohort: cohort,
                updated_at: new Date(),
            }

            let { error } = await supabase.from('profiles').upsert(updates, {
                returning: 'minimal', // Don't return the value after inserting
            })

            if (error)
                throw error
            else
                console.log("Profile Updated successfully")

        } catch (error) {
            console.error(error.message);
        }
    };

    const signup = (email, password1, password2, setMessage) => async e => {
        e.preventDefault();

        // Verify that password is matching
        if (password1 != password2) {
            setMessage("Password does not match");
            console.log("User input mis-matched password");
            return;
        }

        var password = password1;

        if (password.length < 6) {
            setMessage("Password too short (min 6 sharacters)");
            console.log("User password too short");
            return;
        }

        //TODO: hex password here for security
        //note: login also need to hex, when this is implemented

        try {
            setMessage("Signing up ... please be patient...");
            const { error } = await supabase.auth.signUp({ email, password });
            if (error) throw error;
            setMessage("Check your email for the login link!");
            console.log("Verification email sent");
        } catch (error) {
            console.error(error.error_description || error.message);
            setMessage(error.error_description || error.message);
        }
    };

    const login = (email, password) => async e => {
        e.preventDefault();

        //TODO: hex password here for security
        //note: singup also need to hex, when this is implemented

        try {
          const { error } = await supabase.auth.signIn({ email, password });
          if (error) throw error;
            console.log("User logged in");
        } catch (error) {
            alert(error.error_description || error.message);
            console.error(error.error_description || error.message);
        }
    };

    const logout = async e => {
        e.preventDefault();

        try {
            const { error } = await supabase.auth.signOut();
            if (error) throw error;
            console.log("User logged out");
            alert("User logged out");
            
            window.location = window.location.origin.toString();
        } catch (error) {
            console.error(error.error_description || error.message);
        }
    };  

    const sendPasswordReset = (email, setMessage) => async e => {

        e.preventDefault();
        try {
            setMessage("Sending recovery link....please wait")

            const { data, error } = await supabase.auth.api.resetPasswordForEmail(email, {
                redirectTo: `${window.location.origin}/reset-password`,
              });
            if (error) throw error;

            alert("Recovery link has been sent to your email");
            setMessage("Link has been sent!");
        } catch (error) {
            setMessage("Email does not exist");
            console.error(error.error_description);
        }
    };

    const resettingPassword = (password1, password2, setMessage) => async e => {
        e.preventDefault();

        if (password1 != password2) {
            setMessage("Password does not match");
            console.log("User input mis-matched password");
            return;
        }

        var password = password1

        try {
            const {error } = await supabase.auth.update({password});
            if (error) throw error;
            alert("password changed ");
        } catch (error) {
            alert(error.error_description);
            console.error(error.error_description);
        }
    };

    return {
        // Account-releated functions
        signup,
        login,
        logout,
        sendPasswordReset,
        resettingPassword ,
        updateProfileBasic,
        updateProfileAcad,

        // Status of profile info
        profileInfoReady,

        // User-releated info
        email,
        firstName,
        lastName,
        cohort,
        isReady,
        isLocked,
    };
}