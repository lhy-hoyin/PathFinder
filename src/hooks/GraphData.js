import { useState, useContext, createContext } from "react";
import cloneDeep from 'lodash/cloneDeep';
import Color from "color";

import { supabase } from "../supabaseClient";
import { ModuleStateColor } from "../constants"

const graphContext = createContext();

export function ProvideGraphData({ children }) {
    const graphData = useProvideGraphData();
    return <graphContext.Provider value={graphData}>{children}</graphContext.Provider>;
}

export const graphData = () => {
    return useContext(graphContext);
};

function useProvideGraphData() {

    const [modules, setModules] = useState([]);
    const [preq, setPreq] = useState([]);

    const colouring = (selectedColor) => {
        const color ={
           border: Color(selectedColor).darken(0.2).hex(),
           background : selectedColor,
           highlight: {
             border: Color(selectedColor).darken(0.3).hex(),
             background: Color(selectedColor).darken(0.2).hex()
            },
            hover: {
                border: Color(selectedColor).darken(0.3).hex(),
                background: Color(selectedColor).darken(0.2).hex()
            }
        }
         
         return color;
    }

    const orNodesLabel = (orArray) => {
        let label = orArray[0].toString();
        for (var x = 1; x < orArray.length; x++) {
          label = label + " or " + orArray[x]
        }
        return label
    }

    const nodeExist = (checkMod, arr) => {
        let modExist = [];
        let count = 0;
    
        for (var x = 0; x < checkMod.length; x++) {
          if (arr.includes(checkMod[x])) {
            modExist[count] = checkMod[x];
            count++;
          }
        }
        return modExist;
      };

    const updateStatus = (allMods, selectedModule) => {
        let index = allMods.findIndex((x) => x.id === selectedModule);
    
        if (allMods[index].status === true) {
          allMods[index].color = colouring(ModuleStateColor.Completed);
          return allMods;
        }
        //Starting modules that have no prequities
        if (allMods[index].pre.length === 0) {
          if (allMods[index].status === false) {
            allMods[index].color = colouring(ModuleStateColor.Available);
          } else {
            allMods[index].color = colouring(ModuleStateColor.Completed);
          }
        } else {
          //Modules with preq
          for (var count = 0; count < allMods[index].pre.length; count++) {
            
            //For or nodes
            if (allMods[index].pre[count][0].length > 1) {
              for ( var count1 = 0; count1 < allMods[index].pre[count].length; count1++) {
                let temp = allMods.findIndex( (x) => x.id === allMods[index].pre[count][count1]);
                if (allMods[temp].status === true) {
                  allMods[index].color = colouring(ModuleStateColor.Available);
                  return allMods;
                }
              }
              allMods[index].color = colouring(ModuleStateColor.Locked);
              return allMods;
            } else {
              let temp = allMods.findIndex((x) => x.id === allMods[index].pre[count]);

              if (allMods[temp].status === false) {
                allMods[index].color = colouring(ModuleStateColor.Locked);
                return allMods;
              }
            }
          }
    
          allMods[index].color = colouring(ModuleStateColor.Available);
        }
    
        return allMods;
      };

    const getData = (modsArr) => async e => {
        e.preventDefault();

        try {
            let { data, error } = await supabase
                .from("modules")
                .select("*")
                .filter('code', 'in', `(${modsArr})`)

            if (data == null)
                throw ("no data from database")

            let temp = data.slice(0);
            const count = temp.length

            let mod = [];
            let orNodes = [];
            let orCount = 0;

            let edges = [];
            let edgeCount = 0;

            for (var node = 0; node < count; node++) {

                mod[node] = {
                    id: temp[node].code,
                    label: temp[node].code,
                    color: colouring(ModuleStateColor.Completed),
                    info: [temp[node].name, temp[node].acad_year, temp[node].credit, temp[node].description],
                    status: false,
                    pre: []
                };

                if (temp[node].pre_req) {

                    for (var req = 0; req < temp[node].pre_req.length; req++) {
                        
                        let modWithOr = temp[node].pre_req[req].toString().split(",");
                        modWithOr = nodeExist(modWithOr, modsArr)

                        if (modWithOr.length > 1) {

                            orNodes[orCount] = {
                                id:  orNodesLabel(modWithOr),
                                label: orNodesLabel(modWithOr),
                                shape: "ellipse",
                                color: colouring(ModuleStateColor.Completed),
                                info: ["", "", "", ""],
                                status: false,
                                pre: [modWithOr]
                            }
                            orCount++;

                            mod[node].pre.push(orNodesLabel(modWithOr));
                            edges[edgeCount] = { from: temp[node].code, to: orNodesLabel(modWithOr) }
                            edgeCount++;

                            for(var orReq = 0; orReq < modWithOr.length; orReq++) {
                                edges[edgeCount] = { from: orNodesLabel(modWithOr) , to: modWithOr[orReq] }
                                edgeCount++;
                            }

                        } else{
                            if (modWithOr.toString() !== "") {
                                mod[node].pre.push(modWithOr.toString());
                                edges[edgeCount] = { from: temp[node].code, to: modWithOr.toString() };
                                edgeCount++;
                            }
                            
                        }                       
                    }
                }
            }

            mod = mod.concat(orNodes)

            for (var x = 0; x < mod.length; x++) {
                updateStatus(mod, mod[x].id);
            }

            setPreq(edges); // need to set edges first
            setModules(mod);

        } catch (error) {
            alert(error.message);
        }
    };

    const getCourses = async () => {
        try {
            let { data, error } = await supabase
                .from("courses")
                .select("course_name")

            if (data == null)
                throw error 

            return data.map(row => row.course_name)

        } catch (error) {
            console.error(error.message);
        }
    }

    const updateGraph = (nodes) => {
        let index = modules.findIndex((x) => x.id === nodes.toString());
        if (modules[index].color.background === ModuleStateColor.Locked) {
          alert("Module is locked. Please clear the prequities first.");
        } else {
            const modulesCopy = cloneDeep(modules);
            modulesCopy[index].status = !modulesCopy[index].status;
            for (var x = 0; x < modulesCopy.length; x++) {
                updateStatus(modulesCopy, modulesCopy[x].id);
            }
            setModules(modulesCopy);
          //  isUpdating(!updating)
        }
      };

    return {
        getData,
        getCourses,
        updateGraph,

        modules,
        preq
    };
}