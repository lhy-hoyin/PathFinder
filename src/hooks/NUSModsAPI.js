const BASE_URL = "https://api.nusmods.com/v2/";

const generateQueryUrl = (moduleCode, acadYear) => {
	if (moduleCode === undefined) {
		console.error("getMdule: no moduleCode to search")
		return undefined
	}

	if (acadYear === undefined) {
		const currentYear = new Date().getFullYear()
		acadYear = currentYear.toString() + "/" + (currentYear + 1).toString()
	}

	return BASE_URL
		+ acadYear.replace("/", "-").toUpperCase().replace("AY", "")
		+ "/modules/"
		+ moduleCode.toUpperCase()
		+ ".json"
}

export const moduleExist = async (moduleCode, acadYear) => {
	const queryUrl = generateQueryUrl(moduleCode, acadYear) 
	return fetch(queryUrl)
		.then(res => { return res.ok })
};

export const getModule = async (moduleCode, acadYear) => {
	const queryUrl = generateQueryUrl(moduleCode, acadYear) 
	return fetch(queryUrl)
		.then(res => res.json())
		.then(
			(result) => { return { data: result } },
			(err) => { return { error: err } }
		)
};