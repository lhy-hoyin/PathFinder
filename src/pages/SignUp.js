import { useState, useEffect } from "react";
import { useNavigate } from "react-router-dom";

import { Auth } from "../hooks/Auth";
import { supabase } from "../supabaseClient";
import Header from "../components/Header";

import "../css/SignUp.css";

export default function SignUp() {

    const navigate = useNavigate();
    const user = supabase.auth.user();

    const { signup } = Auth();
    const [email, setEmail] = useState('');
    const [password1, setPassword1] = useState('');
    const [password2, setPassword2] = useState('');
    const [message, setMessage] = useState('');

    useEffect(() => {
        if (user != null)
            return navigate("/");
    }, [user]);

    return (
        <>
            <Header />
            <div className="frame">
                <div className="register-new" aria-live="polite">
                    <h1>Sign Up</h1> 
                    <form onSubmit={ signup(email, password1, password2, setMessage) }>
                        <input
                            id="email"
                            className="inputField"
                            type="email"
                            placeholder="Simply register for an account using your email"
                            autoComplete="email"
                            value={email}
                            onChange={(e) => setEmail(e.target.value)} />
                        <input
                            id="password"
                            className="inputField"
                            type="password"
                            placeholder="Password (Minimum 6 characters)"
                            autoComplete="new-password"
                            value={password1}
                            onChange={(e) => setPassword1(e.target.value)} />
                        <input
                            id="confirm-password"
                            className="inputField"
                            type="password"
                            placeholder="Re-enter your password again"
                            autoComplete="off"
                            value={password2}
                            onChange={(e) => setPassword2(e.target.value)} />
                        <button className="button block" aria-live="polite">
                            Register As New User
                        </button>
                    </form>
                </div>
            </div>
            <p>{message}</p>
        </>
    )
}
