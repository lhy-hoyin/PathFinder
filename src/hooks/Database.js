import { supabase } from "../supabaseClient";
import { pullModule, formatPreReq } from "../hooks/NUSModsAPI";

export const getCourseNames = async () => {
    try {
        let { data, error } = await supabase
            .from("courses")
            .select("course_name")

        if (data == null)
            throw error

        return data.map(row => row.course_name)

    } catch (error) {
        console.error(error.message)
    }
};

export const getModuleId = async (modCode, year) => {
    try {
        let { data, status, error } = await supabase
            .from("modules")
            .select("id")
            .eq('acad_year', year)
            .eq('code', modCode.toUpperCase())
            .single()

        if (error && status == 406) {
            /*
             * Error 406: multiple (or no) rows returned
             * No row fround is expected, do nothing here is suppress it
             */
        } else if (data == null)
            throw error

        return data

    } catch (error) {
        console.error(error.message)
        return null
    }
};

export const getModInfo = async (modId) => {
    try {
        if (modId == undefined || modId == null)
            throw { message: "getModInfo: modId is undefined" }

        let { data, error } = await supabase
            .from("modules")
            .select("*")
            .eq('id', modId)
            .single()

        if (data == null)
            throw error

        return data

    } catch (error) {
        console.error(error.message)
    }
};

export const upsertModule = async (moduleCode, acadYear) => {
    try {
        // Retrieve data from API
        const modData = (await pullModule(moduleCode, acadYear)).data

        // Package data properly
        const updates = {
            code: modData.moduleCode,
            name: modData.title,
            description: modData.description,
            acad_year: modData.acadYear,
            credit: modData.moduleCredit,
            preclusion: modData.preclusion,
            pre_req: formatPreReq(modData.prereqTree),
            updated_at: new Date(),
        }

        // if table has existing entry, add the id to update the correct entry
        const existingRowId = await getModuleId(moduleCode, acadYear)
        if (existingRowId)
            updates.id = existingRowId.id

        let { data, error } = await supabase.from('modules').upsert(updates)

        if (error)
            throw error

        console.log("Database Updated:", moduleCode, acadYear)
        return data[0].id

    } catch (error) {
        console.error(error.message);
    }
};

export const getUserAcademic = async (userId) => {
    try {
        let { data, error } = await supabase
            .from("academic")
            .select("*")
            .eq('user_id', userId)

        if (data == null)
            throw error

        return data

    } catch (error) {
        console.error(error.message)
    }
};

export var UserAcademicRecord = {
    insert: async (userId, newModId) => {
        try {
            const { error } = await supabase
                .from('academic')
                .insert([{
                    user_id: userId,
                    module: newModId,
                }])

            if (error)
                throw error
            else
                return { status: 'success' }

        } catch (error) {
            console.error(error.error_description || error.message)
            return {
                status: 'error',
                title: "Oops!",
                description: error.error_description || error.message
            }
        }
    },

    update: async (recordId, isCompleted) => {

        try {
            const { error } = await supabase
                .from('academic')
                .update({ completed: isCompleted })
                .eq('id', recordId)

            if (error)
                throw error
        } catch (error) {
            console.error(error.error_description || error.message);
        }
    },

    delete: async (recordId) => {
        try {
            const { status, error } = await supabase
                .from('academic')
                .delete()
                .eq('id', recordId)
                .single()

            return { status, error }
        } catch (error) {
            console.error(error.message)
            return { status, error }
        }
    },
}