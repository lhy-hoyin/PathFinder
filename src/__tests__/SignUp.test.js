import { render, screen, cleanup, waitFor, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { ChakraProvider } from '@chakra-ui/react';
import { BrowserRouter } from "react-router-dom";
import { ProvideAuth } from "../hooks/Auth";
import SignUp from "../pages/SignUp";

beforeEach(() => {
    render(
        <ChakraProvider>
            <ProvideAuth>
                <BrowserRouter>
                    <SignUp />
                </BrowserRouter>
            </ProvideAuth>
        </ChakraProvider>
    );
});

afterEach(cleanup);

it("renders without crashing", () => {
    expect(renderer.create(
        <ChakraProvider>
            <ProvideAuth>
                <BrowserRouter>
                    <SignUp />
                </BrowserRouter>
            </ProvideAuth>
        </ChakraProvider>
    ).toJSON()).toMatchSnapshot();
});

it("text display correctly", () => {
    expect(screen.getByRole("heading", { name:"Sign Up"})).toBeDefined()

    expect(screen.getByPlaceholderText("Email")).toBeDefined()

    expect(screen.getByPlaceholderText("Enter Password")).toBeDefined()
    expect(screen.getByText("Show")).toBeDefined()

    expect(screen.getByPlaceholderText("Re-enter Password")).toBeDefined()

    expect(screen.getByText("Register As New User")).toBeDefined()
})

test("toggle show password function", async () => {
    const passwordInput = screen.getByPlaceholderText("Enter Password")
    const passwordToggle = screen.getByText("Show")

    expect(passwordInput.type).toBe("password")

    await waitFor(() => {
        fireEvent(passwordToggle,
            new MouseEvent('click', { bubbles: true, cancelable: true, })
        )
        expect(passwordInput.type).toBe("text")
    })

    await waitFor(() => {
        fireEvent(passwordToggle,
            new MouseEvent('click', { bubbles: true, cancelable: true, })
        )
        expect(passwordInput.type).toBe("password")
    })
})