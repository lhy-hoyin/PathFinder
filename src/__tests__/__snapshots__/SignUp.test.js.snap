// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  <div
    className="header"
  >
    <a
      className="chakra-link css-spn4bz"
      href="/"
    >
      <img
        alt="Pathfinder"
        className="chakra-image css-73nay0"
        src="img/banner.png"
      />
    </a>
    <div
      className="nav-links"
    >
      <ul>
        <li
          className="clickable"
        >
          <a
            href="/sign-up"
          >
            Sign Up
          </a>
        </li>
        <li
          className="clickable"
        >
          <a
            onClick={[Function]}
          >
            Login
          </a>
        </li>
      </ul>
    </div>
  </div>,
  <div
    className="frame"
  >
    <div
      aria-live="polite"
      className="register-new"
    >
      <form
        onSubmit={[Function]}
      >
        <h2
          className="chakra-heading css-1dklj6k"
        >
          Sign Up
        </h2>
        <p
          className="chakra-text css-0"
        >
          Create An 
          <i>
            PathFinder
          </i>
           account now using your email and password
        </p>
        <div
          className="chakra-form-control css-1kxonj9"
          role="group"
        >
          <input
            aria-required={true}
            autoComplete="email"
            className="chakra-input css-14sef68"
            disabled={false}
            id="email"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Email"
            readOnly={false}
            required={true}
            type="email"
            value=""
          />
          <div
            className="chakra-input__group css-86lbzn"
          >
            <input
              aria-required={true}
              autoComplete="new-password"
              className="chakra-input css-1fjljm8"
              disabled={false}
              id="password"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Enter Password"
              readOnly={false}
              required={true}
              type="password"
              value=""
            />
            <div
              className="chakra-input__right-element css-ohd79a"
              size="md"
            >
              <button
                className="chakra-button css-19wuo85"
                onClick={[Function]}
                type="button"
              >
                Show
              </button>
            </div>
          </div>
          <div
            className="chakra-form__helper-text css-11mx7yv"
            id="field-:r2:-helptext"
          >
            Password should contain at least 6 characters
          </div>
          <input
            aria-required={true}
            autoComplete="off"
            className="chakra-input css-14sef68"
            disabled={false}
            id="confirm-password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Re-enter Password"
            readOnly={false}
            required={true}
            type="password"
            value=""
          />
          <div
            className="chakra-form__helper-text css-11mx7yv"
            id="field-:r2:-helptext"
          >
            You can't show/hide this, so make sure to type correctly
          </div>
          <button
            className="chakra-button css-eu2w2k"
            disabled={false}
            type="submit"
          >
            Register As New User
          </button>
        </div>
      </form>
      <p />
    </div>
  </div>,
  <span />,
  <span
    className="chakra-env"
    hidden={true}
  />,
]
`;
